pkg/operator/ceph/cluster/controller.go

```diff
@@ -1025,28 +1025,33 @@ func (c *ClusterController) handleDelete(cluster *cephv1.CephCluster, retryInter
 }
 
 func purgeExternalCluster(clientset kubernetes.Interface, namespace string) error {
-	// purge the mon endpoint config map
-	err := clientset.CoreV1().ConfigMaps(namespace).Delete(mon.EndpointConfigMapName, &metav1.DeleteOptions{})
-	if err != nil && !kerrors.IsNotFound(err) {
-		return err
+	// Purge the config maps
+	cmsToDelete := []string{
+		mon.EndpointConfigMapName,
+		config.StoreName,
+		k8sutil.ConfigOverrideName,
+	}
+	for _, cm := range cmsToDelete {
+		err := clientset.CoreV1().ConfigMaps(namespace).Delete(cm, &metav1.DeleteOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to delete config map %+v", cm)
```

Comment:
Do we really want to return, or should we just log the error and continue attempting to clean everything else up?

Reply:
Looks like the caller will just log the error and not ever try again to cleanup, so seems like we shouldn't return here so we can attempt to delete all resources.